lib/ukstore/README.md

```diff
@@ -0,0 +1,189 @@
+    obj = uk_store_acquire_object(data->library_id, data->object_id);
+    if (unlikely(!obj))
+        return UK_EVENT_HANDLED; /* The object was already released */
+
+    /* At this point we hold a valid pointer of the object. It won't be
+     * freed until we decrement the refcount.
+     */
+    entry = uk_store_get_entry(UK_MYDEVICE_LIB_ID, obj->id, UK_MYDEV_METRICS_IRQ_COUNT);
+
+    /* Create /sys/mydevice/irq_count and associate this entry with that file.
+     * Call `uk_store_get_value()` every time the user reads from /sys/mydevice/irq_count.
+     */
+
+    return UK_EVENT_HANDLED;
+}
+
+/* Registers handler */
+UK_EVENT_HANDLER_(UK_STORE_EVENT_CREATE_OBJECT, handle_create_object);
+```
+
+#### Device Termination: Producer
+```c
+/* Called by the driver on device termination */
+int uk_mydevice_teardown_metrics(struct uk_mydevice *dev)
+{
+    struct uk_store_object *obj;
+
+    /* Is this coming from our library? */
+    if (data->library_id != UK_MYDEVICE_LIB_ID)
+        return UK_EVENT_NOT_HANDLED; /* continue to next handler */
+
+    /* this is looked up via library-specific means */
+    obj = get_obj_by_device(dev);
+
+    /* Notify consumers */
+    event_data = () {
+        .library_id = UK_MYDEVICE_LIB_ID,
+        .object_id = obj->id,
+        .owner = NULL
+    };
+    uk_raise_event(UK_STORE_EVENT_RELEASE_OBJECT, &event_data);
+
+    /* Decrement the refcount. The object will be freed as soon as
+     * all consumers decrement the refcount.
+     */
+    uk_store_release_object(UK_MYDEVICE_LIB_ID, obj);
+
+    return 0;
+}
+```
+
+#### Device Termination: Consumer
+
+```c
+/* This is the handler for UK_STORE_EVENT_RELEASE_OBJECT */
+static int handle_release_object(struct uk_event_data *data)
+{
+    struct uk_store_event_data *data = (struct uk_store_event_data *)arg;
+
+    /* Use library-specific means to stop pulling values
+     * and remove /sys/mydevice/irq_count.
+     */
+
+    /* Decrement the refcount. If we're the last ones the object will be
+     * free, along with its entries.
+     */
+    uk_store_release_object(UK_MYDEVICE_LIB_ID, obj);
+}
+
+/* Registers handler */
+UK_EVENT_HANDLER_(UK_STORE_EVENT_RELEASE_OBJECT, handle_release_object);
+```
+
+## Static API
+
+The static API allows libraries to define a flat list of entries, ie without parent object.
+The hierarchy is then defined as `library/entry[value]`.
+Static entries are at compile-time via the `UK_STORE_STATIC_ENTRY` macro.
+
+The static API is a reduced version of the dynamic API as:
+  * There is no requirement for dynamic memory allocation.
+  * There is no reference counting.
+  * uk_store events are not used.
+  * Consumers access entries directly, without the need to call `uk_store_acquire_object()`. Calling `uk_store_get_entry()` requires passing `UK_STORE_OBJECT_ID_NONE` to the `object_id` parameter.
+
+The static API is useful inthe cases where a library exports a fixed set of entries that do not depend on individual instances.
```

Comment:
"The static API is useful `in the` cases"

Reply:
Done